### PR TITLE
Doing dh

### PR DESCRIPTION
**Pull request by danteg93**
mhm

### CODE REVIEW

File: SudokuProject/src/cspSolver/BTSolver.java [hunk 1]
```diff
@@ -243,6 +244,7 @@ private Variable getfirstUnassignedVariable()
 		{
 			if(!v.isAssigned())
 			{
+				System.out.println("Returning var " + v.toString());
```
Comment (Tybot204) [inline SudokuProject/src/cspSolver/BTSolver.java]: Rogue println still here.

Comment (danteg93) [inline SudokuProject/src/cspSolver/BTSolver.java]: SHIEEET

File: SudokuProject/src/cspSolver/BTSolver.java [hunk 2]
```diff
@@ -277,6 +279,59 @@ private Variable getMRV()
...
+	 */
+	private int getDegree(Variable v){
+		int degree = 0;
+		for(Variable w: network.getNeighborsOfVariable(v)){
+			if(!w.isAssigned()){
+				degree++;
+			}
+		}
+		return degree;
+	}
```
Comment (Tybot204) [inline SudokuProject/src/cspSolver/BTSolver.java:309]: This is purely just code DRYness but there's probably a method we can call on the array returned by `getNeighborsOfVariable(v)` to count how many objects have a certain condition false.

EDIT: Damn just looked it up and it's not worth it. The simple `.filter` method is new to Java 8 so we can't use it.

File: SudokuProject/src/cspSolver/BTSolver.java [hunk 3]
```diff
@@ -277,6 +278,59 @@ private Variable getMRV()
...
+	 */
+	private Variable getMRVWithDH()
+	{
+		List<Variable> tempNetwork = network.getVariables();
+		int minLegalMoves = Integer.MAX_VALUE;
+		Variable tempVar = tempNetwork.get(0);
+		for(Variable v: tempNetwork){
+			if(!v.isAssigned())
+			{
+				if(v.size() == minLegalMoves && getDegree(v) > getDegree(tempVar)){
```
Comment (Tybot204) [inline SudokuProject/src/cspSolver/BTSolver.java]: We should keep track of the current degree of `tempVar` in a variable so we don't have to loop through all neighbors every time here.

File: SudokuProject/src/cspSolver/BTSolver.java [hunk 4]
```diff
@@ -277,6 +278,59 @@ private Variable getMRV()
...
+					tempVar = v;
+				}
+				else if(v.size() < minLegalMoves){
+					minLegalMoves = v.size();
+					tempVar = v;
```
Comment (Tybot204) [inline SudokuProject/src/cspSolver/BTSolver.java:327]: We would just assign that temp degree variable to `getDegree(v)` here.

**Pull request merged**